### ISSUE TITLE
cmd/go: TestRepoRootForImportPath fails

### ISSUE THREAD

**Issue by josharian** [golang/go#15979]
At tip:

```bash
$ go test cmd/go
--- FAIL: TestRepoRootForImportPath (0.92s)
	vcs_test.go:159: RepoRootForImport("git.openstack.org/openstack/swift.git"): invalid version control suffix in git.openstack.org path
FAIL
FAIL	cmd/go	207.939s
```

This test doesn't run in `-short` mode, which is probably why no one noticed. (We could really use a builder that runs the full tests.)

**Comment by ianlancetaylor** [comment]
As far as I can see, this test has been broken since it was submitted in https://golang.org/cl/23135, but nobody noticed because the test does not run with -test.short as used on the builders.

I asked on the CL what the intent is.

**Comment by emonty** [comment]
Bother. I'll jump on this right away - sorry.

**Comment by josharian** [comment]
Any status update, @emonty?

**Comment by adg** [comment]
Should be fixed by afccfb829f17f85a8d4aa74f1a11a27422437405.
